### PR TITLE
Open source build improvements

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Make several improvements to Jenkins and community building:

1) If we're running on a non-internal-developer machine, just don't try to deploy extensions since that'll always fail
2) Remove the CIBuild flag entirely -- community and Jenkins builds should just do the same thing
3) Run the same tests everywhere

Reviewers: @jaredpar, @amcasey, @davkean, @tannergooding, @agocke

### CODE REVIEW

**Comment by jaredpar** [comment]
:+1: 

Build seems to have failed with the following:

``` 
"d:\j\workspace\dotnet_roslyn_windows_prtest\src\Samples\..\..\packages\NuGet.CommandLine.2.8.5\tools\NuGet.exe pack Diagnostic.nuspec -NoPackageAnalysis -OutputDirectory d:\j\workspace\dotnet_roslyn_windows_prtest\Binaries\Debug\" exited with code 3.
  MakeConstCS.UnitTests -> 
````

Looks unrelated.  Will retest.

**Comment by jaredpar** [comment]
@dotnet-bot  test this please

**Comment by jaredpar** [comment]
Hmm, failure looks real.

**Comment by tannergooding** [comment]
@dotnet-bot test this please

Going to try and repro it on the build machine.

**Comment by tannergooding** [comment]
@dotnet-bot test this please

**Comment by jaredpar** [comment]
@tannergooding the problem is this line in the build file:

> https://github.com/dotnet/roslyn/blob/master/src/Samples/CSharp/AsyncPackage/AsyncPackage.csproj#L194

@jasonmalinowski removed the `CIBuild` option hence this is getting triggered on the build server where it can't run correctly.

**Comment by jasonmalinowski** [comment]
I removed the check in AsyncPackage.csproj because once I cleaned up everything else that was the *only* remaining thing depending upon CIBuild. I'll investigate further what's up.

**Comment by tmeschter** [comment]
@jasonmalinowski I already have a pull request open to remove AsyncPackage entirely, as I've already moved to the dotnet/roslyn-analyzers.

**Comment by jasonmalinowski** [comment]
@tmeschter, when do you anticipate that'll get merged in?

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 1]
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
                     File.Delete(resultsPath);
                 }
 
-                errorOutput = processOutput.ErrorLines.Any()
-                    ? processOutput.ErrorLines.Aggregate((x, y) => x + Environment.NewLine + y)
-                    : string.Format("xunit produced no error output but had exit code {0}", processOutput.ExitCode);
+                errorOutput.AppendLine($"Command: {_xunitConsolePath} {builder.ToString()}");
```
Comment (amcasey) [inline src/Tools/Source/RunTests/TestRunner.cs]: Isn't ```.ToString()``` implied?

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 2]
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
...
 
-                errorOutput = string.Format("Command: {0} {1}", _xunitConsolePath, builder.ToString())
-                    + Environment.NewLine
-                    + errorOutput;
+                if (processOutput.ErrorLines.Any())
```
Comment (amcasey) [inline src/Tools/Source/RunTests/TestRunner.cs]: Extract local for ```processOutput.ErrorLines```?

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 3]
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
...
+                {
+                    foreach (var line in processOutput.ErrorLines)
+                    {
+                        errorOutput.AppendLine(line);
+                    }
+                }
+                else
+                {
+                    errorOutput.Append($"xunit produced no error output but had exit code {processOutput.ExitCode}");
```
Comment (amcasey) [inline src/Tools/Source/RunTests/TestRunner.cs]: ```AppendLine```?

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 4]
```diff
@@ -155,13 +155,19 @@ private void Print(List<TestResult> testResults)
                     File.Delete(resultsPath);
                 }
 
-                errorOutput = processOutput.ErrorLines.Any()
-                    ? processOutput.ErrorLines.Aggregate((x, y) => x + Environment.NewLine + y)
```
Comment (amcasey) [inline src/Tools/Source/RunTests/TestRunner.cs]: This is hilarious.

**Comment by amcasey** [comment]
LGTM, assuming you tested the three cases (Roslyn dev, external dev, Jenkins).

**Comment by jasonmalinowski** [comment]
Jenkins will be fixed once @tmecshter deletes the thing causing me pain. I've verified BuildAndTest works on a clean box without dogfood tools (although I needed to disable strong name verification to get around the test resources problem). Once those are both fixed I'll also do the internal dev test too.

**Comment by jasonmalinowski** [comment]
And @tmeschter as removed the async analyzers, so I've rebased and we'll see what Jenkins thinks now.

**Comment by jaredpar** [comment]
@dotnet-bot test this please 

One more Jenkins run as we changed MSBuild cross plat at the same time.

**Pull request merged**